pkg/storage/cachingtempstore.go

```diff
@@ -121,6 +121,13 @@ func (ttrw *CachingTempStore) Put(ctx context.Context, key string, data []byte)
 
 // Close will clean up any temporary resources used by the storage.
 func (ttrw *CachingTempStore) Close() error {
+	// we need to ensure that the writer receives no more data, so swap
+	// it out with a no-op writer that returns an error
+	ttrw.store.lk.Lock()
+	ttrw.outWriter = func(lc linking.LinkContext) (io.Writer, linking.BlockWriteCommitter, error) {
+		return nil, nil, errClosed
+	}
```

Comment:
Do we know what the behavior of returing `errClosed` results in? I tried following where it bubbles up that error and got lost in the `RetrievalRequest`.

Reply:
it'll at worst go into a log. there's normally not much to do at the point that you're calling `.Close()`